### ISSUE TITLE
Bug when we quit/rejoin the server

### ISSUE THREAD

**Issue by Odasio** [Arckenver/Nations#156]
Hi there ! We use your last version of nations 2.5, on a sponge server with mods in 1.10.2. We had a little issue, when we restart our server all our memberslist of nations are wiped and noone are in those files. In the same way, when we quit the server it kicks us of the nation admin!

 Could you please help us ?

**Comment by TheoKah** [comment]
If I understand correctly: the nations are still there, only the members are removed?

**Comment by MortelKni** [comment]
It's that. After each reboot nations are still there, with their config, but there is no member inside.
I checked the files of the nations, and it seems that the players are not registered at all.

**Comment by TheoKah** [comment]
Are the members the only thing not saved? Are the claimed blocks, zones and flags saved? Do you have an error message in console?

**Comment by MortelKni** [comment]
Yes, everything except the members are properly saved. At the moment I did not noticed any error message in the console !

**Comment by TheoKah** [comment]
Only reason citizens would be lost is if this was an admin nation. Admin nations will not save citizens, ministers, president, taxes, extra bocks and extra spawns.

EDIT: I now reread your message and see you mention it was "nation admin". I understood that as "nation president", thus my confusion.

**Comment by Odasio** [comment]
Ah ! That's it ! And there is no way they will save citizens ? 

Another point, in those nations admins (and zones inside them), our players whom aren't in those zone can't place nor interact with boats and citizens can't outside the zones use their bows !

**Comment by TheoKah** [comment]
Well I can make it so the admin nations save their citizens. However I see no point in doing it as it was though as only admins with `/na` perms can mess with it.

If you want people to be able to interact in the whole nation, you can set the "interact" permission for "outsiders" to true (either with `/na perm` or with `/n info NATION` then click on the correct permission).
Same with zones.

The bow is blocked by the PvP flag, turn is on and should do the trick.

If you tell me more about what you want to do, I can change a few things in the plugin if necessary.
If you don't want to explain publicly, we can use discord.

**Comment by Odasio** [comment]
In fact, we use the /na to make easier for the mods (who are the ruler of the nations) the possibility to leave and create other role-play nations. So we're obliged to !

For the interaction, we don't want the outsider to have the possibility to open doors, chest and all, but just to take, use and destroy entities like boats (to have PVP raids). If this is not possible I would understand !

And for the bow, that's kinda awkward no ? They can't use this for PVM nor the PVP, but it's blocked only for the PVP in the /n info nations !

If you don't understand me completely i would drop by the discord ahah

**Comment by TheoKah** [comment]
The bow can be used if one has the 'interact' permission. But the arrow cannot hurt a player if pvp is off. (Should be able to hurt monsters, passives mobs require the build permission (they are considered as part of the furniture) but this last part is about to change)

Boats are considered passives entities iirc, so this might need the build perm as well.

I can see your point. I'll provide a better way for you to handle this. I'll see what can be done next week end. 

I'll also make it possible to save citizens in admin nations.

**Comment by Odasio** [comment]
That super sweet of you !

I'll stay tune for the next week-end so ! You think my requests are realisable ?

**Comment by TheoKah** [comment]
I still need to think about it. I think I'll change the flag system. Add more flags... Something like that.

**Comment by Odasio** [comment]
Thank you for thinking about it anyway !

**Comment by Arckenver** [comment]
Nations has been completely re-written in Kotlin since [version v3.0](https://github.com/Arckenver/Nations/releases/tag/v3.0), I'm therefore closing all issues opened prior to `v3.0`.

Please, feel free to re-create a new issue if the topic is still relevant.
